app/src/main/java/org/simple/clinic/enterotp/EnterOtpScreen.kt

```diff
@@ -0,0 +1,65 @@
+package org.simple.clinic.enterotp
+
+import android.content.Context
+import android.support.v7.widget.Toolbar
+import android.util.AttributeSet
+import android.widget.RelativeLayout
+import android.widget.TextView
+import com.jakewharton.rxbinding2.view.RxView
+import io.reactivex.Observable
+import io.reactivex.android.schedulers.AndroidSchedulers.mainThread
+import io.reactivex.schedulers.Schedulers
+import kotterknife.bindView
+import org.simple.clinic.R
+import org.simple.clinic.activity.TheActivity
+import org.simple.clinic.router.screen.ScreenRouter
+import org.simple.clinic.widgets.PinEditText
+import org.simple.clinic.widgets.showKeyboard
+import javax.inject.Inject
+
+class EnterOtpScreen(context: Context, attributeSet: AttributeSet) : RelativeLayout(context, attributeSet) {
+
+  @Inject
+  lateinit var controller: EnterOtpScreenController
+
+  @Inject
+  lateinit var screenRouter: ScreenRouter
+
+  private val userPhoneNumberTextView by bindView<TextView>(R.id.enterotp_phonenumber)
+  private val otpEntryEditText by bindView<PinEditText>(R.id.enterotp_otp)
+  private val toolBar by bindView<Toolbar>(R.id.enterotp_toolbar)
+
+  override fun onFinishInflate() {
+    super.onFinishInflate()
+    if (isInEditMode) {
+      return
+    }
+    TheActivity.component.inject(this)
+
+    screenCreates()
+        .observeOn(Schedulers.io())
+        .compose(controller)
+        .observeOn(mainThread())
+        .takeUntil(RxView.detaches(this))
+        .subscribe { uiChange -> uiChange(this) }
+
+    // The library does not seem to pick up the max length from
+    // the attribute in the layout, so we set it manually
+    otpEntryEditText.setMaxLength(6)
```

Comment:
Isn't `PinEditText` being used for entering the PIN as well, which has a length of 4? Are you suggesting we have a separate `OtpEditText` which sets the max length internally to 6?

Reply:
Sorry I did not mean harcoding `setMaxLength(6)` inside `PinEditText`. But maybe it can call `setMaxLength(maxLength)` again inside its constructor to force the text length?

Reply:
Call me if this isn't clear :)

Reply:
@saket I tried something out, and it looks like it picks up the `maxLength` attribute **IF** it is set on the widget in the layout directly instead of the style. I will change all usages of this widget to specify the length on the widget directly.